File: JS/BOJ-S3-2559-Progression.js
"use strict";

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.trimEnd().split(' '));

let output = [];
const solution = (input) => {
  const [N, K] = input[0].map(v => +v);
  const nums = input[1].map(v => +v);
  const psum = Array.from({length: N+1}, ()=>0);
  let maxVal = -1e9;

  for(let i = 1; i <= N; ++i) {
    psum[i] = psum[i-1] + nums[i-1];
  }
  for(let i = K; i <= N; ++i){
    maxVal = Math.max(maxVal, psum[i] - psum[i-K]);
  }

  output.push(maxVal);
};

solution(input);
fs.writeFileSync("/dev/stdout", output.join("").trimEnd());
// fs.writeFileSync("../output.txt", output.join("").trimEnd(), (err) => console.error);

File: CPP/BOJ-S3-2559-Progression.cpp
#include <bits/stdc++.h>
using namespace std;

int n, k, temp, psum[100001], ret = -10000004;

int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  // freopen("input.txt", "r", stdin);
  // freopen("output.txt", "w", stdout);

  cin >> n >> k;
  for (int i = 1; i <= n; ++i)
  {
    cin >> temp;
    psum[i] = psum[i - 1] + temp;
  }

  for (int i = k; i <= n; ++i)
  {
    ret = max(ret, psum[i] - psum[i - k]);
  }

  cout << ret << '\n';

  return 0;
}